File: LeetCode/67-add-binary/solution.ts
/**
 * Adds two binary strings and returns their sum as a binary string.
 *
 * @param {string} a - The first binary string.
 * @param {string} b - The second binary string.
 * @returns {string} - The sum of the two binary strings as a binary string.
 */
function addBinary(a: string, b: string): string {
  // Convert both binary string inputs into BigInt using the "0b" prefix
  const sum = BigInt("0b" + a) + BigInt("0b" + b);
  // Convert the resulting BigInt sum back to a binary string and return it
  return sum.toString(2);
};

File: LeetCode/67-add-binary/Solution.java
class Solution {
  /**
   * Adds two binary strings and returns their sum as a binary string.
   *
   * @param a The first binary string.
   * @param b The second binary string.
   * @return The sum of the two binary strings as a binary string.
   */
  public String addBinary(String a, String b) {
    return Integer.toBinaryString(Integer.parseInt(a, 2) + Integer.parseInt(b, 2));
  }
}